atlp-rwanda/e-commerce-mavericks-fn
# [finishes #187816382] product details page

Here's a structured template you can use for your pull request (PR):

---

## Purpose

The purpose of this PR is to enhance the product viewing page by implementing features that allow users to view detailed information about a product, including recommended products, reviews, available colors, and sizes. Additionally, this PR introduces functionalities for adding a product to the wishlist and the cart.

## Changes Made

- Implemented detailed product view including:
  - Display of recommended products
  - Display of product reviews
  - Display of available colors and sizes
- Added functionality for users to:
  - Add products to their wishlist
  - Add products to their cart
- Updated UI components to accommodate the new features
- Refactored existing code to ensure seamless integration of new functionalities

## Testing Instructions

1. Navigate to the product page.
2. Verify that the product details (including recommended products, reviews, available colors, and sizes) are displayed correctly.
3. Test adding the product to the wishlist by clicking the "Add to Wishlist" button.
4. Test adding the product to the cart by clicking the "Add to Cart" button.
5. Ensure that the product is correctly added to both the wishlist and the cart, and that the respective counts are updated.
6. Run all unit and integration tests to verify that the new features do not break existing functionality.

## Related Issues

- Issue #123: Implement product detail view
- Issue #124: Add product to wishlist functionality
- Issue #125: Add product to cart functionality

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:


---

![image](https://github.com/atlp-rwanda/e-commerce-mavericks-fn/assets/101561925/3c245358-7c21-40bb-9ac9-4a9efb4a1d1c)
